src/elfpy/pricing_models/yieldspace.py

```diff
@@ -56,42 +49,49 @@ def calc_lp_out_given_tokens_in(
 
         """
         assert d_base > 0, f"pricing_models.calc_lp_out_given_tokens_in: ERROR: expected d_base > 0, not {d_base}!"
-        assert (
-            share_reserves >= 0
-        ), f"pricing_models.calc_lp_out_given_tokens_in: ERROR: expected share_reserves >= 0, not {share_reserves}!"
-        assert (
-            bond_reserves >= 0
-        ), f"pricing_models.calc_lp_out_given_tokens_in: ERROR: expected bond_reserves >= 0, not {bond_reserves}!"
-        assert (
-            base_buffer >= 0
-        ), f"pricing_models.calc_lp_out_given_tokens_in: ERROR: expected base_buffer >= 0, not {base_buffer}!"
-        assert (
-            lp_reserves >= 0
-        ), f"pricing_models.calc_lp_out_given_tokens_in: ERROR: expected lp_reserves >= 0, not {lp_reserves}!"
+        assert market_state.share_reserves >= 0, (
+            "pricing_models.calc_lp_out_given_tokens_in: ERROR:  "
+            f"Expected share_reserves >= 0, not {market_state.share_reserves}!"
+        )
+        assert market_state.bond_reserves >= 0, (
+            "pricing_models.calc_lp_out_given_tokens_in: ERROR: "
+            f"Expected bond_reserves >= 0, not {market_state.bond_reserves}!"
+        )
+        assert market_state.base_buffer >= 0, (
+            "pricing_models.calc_lp_out_given_tokens_in: ERROR: "
+            f"Expected base_buffer >= 0, not {market_state.base_buffer}!"
+        )
+        assert market_state.lp_reserves >= 0, (
+            "pricing_models.calc_lp_out_given_tokens_in: ERROR: "
+            f"Expected lp_reserves >= 0, not {market_state.lp_reserves}!"
+        )
         assert rate >= 0, f"pricing_models.calc_lp_out_given_tokens_in: ERROR: expected rate >= 0, not {rate}!"
-        assert (
-            1 > time_remaining >= 0
-        ), f"pricing_models.calc_lp_out_given_tokens_in: ERROR: expected 1 > time_remaining >= 0, not {time_remaining}!"
-        assert stretched_time_remaining >= 0, (
+        assert 1 > time_remaining.normalized_time >= 0, (
+            "pricing_models.calc_lp_out_given_tokens_in: ERROR: "
+            f"expected 1 > time_remaining >= 0, not {time_remaining.normalied_time}!"
+        )
+        assert time_remaining.stretched_time >= 0, (
```

Comment:
We should probably encapsulate sanity checks within our time structures so that we don't need to run the asserts every time a time value is used (I realize that this may make things harder during the refactor of time in the simulator, so I'd encourage us to punt on this for now).